
Remove compiler warnings for gcc 4.5.0

init/main.c:734:10: warning: 'calltime.tv64' may be used uninitialized in
	this function
kernel/async.c:127:10: warning: 'calltime.tv64' may be used uninitialized in
	this function
kernel/async.c:283:10: warning: 'starttime.tv64' may be used uninitialized in
	this function
drivers/base/power/main.c:305:10: warning: 'calltime.tv64' may be used uninitialized
	in this function
drivers/base/power/main.c:305:10: warning: 'calltime.tv64' may be used uninitialized
	in this function
fs/eventpoll.c:1118:7: warning: 'slack' may be used uninitialized in this function
drivers/md/dm.c:1118:3: warning: statement with no effect
drivers/md/dm.c:1146:3: warning: statement with no effect
drivers/md/dm-table.c:1065:2: warning: statement with no effect
drivers/md/dm-table.c:1071:2: warning: statement with no effect
net/ipv4/netfilter/nf_nat_standalone.c:118:2: warning: case value '4' not
	in enumerated type 'enum ip_conntrack_info'
net/netfilter/nf_conntrack_netlink.c:762:11: warning: comparison between
	'enum ctattr_tuple' and 'enum ctattr_type'
lib/decompress_bunzip2.c:511:1: warning: the frame size of 1872 bytes is
	larger than 1024 bytes
drivers/regulator/core.c:2606:17: warning: 'err_ptr' may be used uninitialized
	in this function
drivers/media/video/gspca/gspca.c:1508:2: warning: case value '7' not in
	enumerated type 'enum v4l2_memory'

Change-Id: I72bd4784836c1f18d643b0747a296a2660106fa9
Signed-off-by: Ezra Godfrey <egodfrey@codeaurora.org>
---

diff --git a/arch/arm/mm/Makefile b/arch/arm/mm/Makefile
index e28730e..8f012ca 100644
--- a/arch/arm/mm/Makefile
+++ b/arch/arm/mm/Makefile
@@ -104,3 +104,6 @@ obj-$(CONFIG_CACHE_TAUROS2)	+= cache-tauros2.o
 
 obj-$(CONFIG_VCM)		+= vcm.o vcm_alloc.o
 obj-$(CONFIG_VCM_MM)		+= vcm_mm.o
+
+# Ignore warnings to prevent build breaks when using gcc 4.5.0
+CFLAGS_REMOVE_alignment.o = -Werror
diff --git a/drivers/base/power/Makefile b/drivers/base/power/Makefile
index 89de753..029128e 100644
--- a/drivers/base/power/Makefile
+++ b/drivers/base/power/Makefile
@@ -6,3 +6,6 @@ obj-$(CONFIG_PM_TRACE_RTC)	+= trace.o
 
 ccflags-$(CONFIG_DEBUG_DRIVER) := -DDEBUG
 ccflags-$(CONFIG_PM_VERBOSE)   += -DDEBUG
+
+# Ignore warnings to prevent build breaks when using gcc 4.5.0
+CFLAGS_REMOVE_main.o = -Werror
diff --git a/drivers/md/Makefile b/drivers/md/Makefile
index e355e7f..61c967e 100644
--- a/drivers/md/Makefile
+++ b/drivers/md/Makefile
@@ -117,3 +117,7 @@ quiet_cmd_mktable = TABLE   $@
 targets += raid6tables.c
 $(obj)/raid6tables.c: $(obj)/mktables FORCE
 	$(call if_changed,mktable)
+
+# Ignore warnings to prevent build breaks when using gcc 4.5.0
+CFLAGS_REMOVE_dm.o = -Werror
+CFLAGS_REMOVE_dm-table.o = -Werror
diff --git a/drivers/media/video/gspca/Makefile b/drivers/media/video/gspca/Makefile
index 6e4cf1c..412e609 100644
--- a/drivers/media/video/gspca/Makefile
+++ b/drivers/media/video/gspca/Makefile
@@ -71,3 +71,6 @@ gspca_zc3xx-objs    := zc3xx.o
 obj-$(CONFIG_USB_M5602)   += m5602/
 obj-$(CONFIG_USB_STV06XX) += stv06xx/
 obj-$(CONFIG_USB_GL860)   += gl860/
+
+# Ignore warnings to prevent build breaks when using gcc 4.5.0
+CFLAGS_REMOVE_gspca.o = -Werror
diff --git a/drivers/regulator/core.c b/drivers/regulator/core.c
index 025bde2..2971fde 100644
--- a/drivers/regulator/core.c
+++ b/drivers/regulator/core.c
@@ -2603,7 +2603,7 @@ static int reg_debug_init(void)
 static int regulator_debug_create_directory(struct regulator_dev *regulator_dev)
 {
 	struct dentry *reg_subdir;
-	struct dentry *err_ptr;
+	struct dentry *err_ptr = NULL;
 	struct regulator *reg;
 	struct regulator_ops *reg_ops;
 	mode_t mode;
diff --git a/fs/Makefile b/fs/Makefile
index e872948..2a466c7 100644
--- a/fs/Makefile
+++ b/fs/Makefile
@@ -129,3 +129,6 @@ obj-$(CONFIG_CEPH_FS)		+= ceph/
 
 # Patched by YAFFS
 obj-$(CONFIG_YAFFS_FS)		+= yaffs2/
+
+# Ignore warnings to prevent build breaks when using gcc 4.5.0
+CFLAGS_REMOVE_eventpoll.o = -Werror
diff --git a/init/Makefile b/init/Makefile
index 0bf677a..0885f60 100644
--- a/init/Makefile
+++ b/init/Makefile
@@ -30,3 +30,6 @@ include/generated/compile.h: FORCE
 	@$($(quiet)chk_compile.h)
 	$(Q)$(CONFIG_SHELL) $(srctree)/scripts/mkcompile_h $@ \
 	"$(UTS_MACHINE)" "$(CONFIG_SMP)" "$(CONFIG_PREEMPT)" "$(CC) $(KBUILD_CFLAGS)"
+
+# Ignore warnings to prevent build breaks when using gcc 4.5.0
+CFLAGS_REMOVE_main.o = -Werror
diff --git a/kernel/Makefile b/kernel/Makefile
index 057472f..908095b 100644
--- a/kernel/Makefile
+++ b/kernel/Makefile
@@ -25,6 +25,9 @@ CFLAGS_REMOVE_sched_clock.o = -pg
 CFLAGS_REMOVE_perf_event.o = -pg
 endif
 
+# Ignore warnings to prevent build breaks when using gcc 4.5.0
+CFLAGS_REMOVE_async.o = -Werror
+
 obj-$(CONFIG_FREEZER) += freezer.o
 obj-$(CONFIG_PROFILING) += profile.o
 obj-$(CONFIG_SYSCTL_SYSCALL_CHECK) += sysctl_check.o
diff --git a/net/ipv4/netfilter/Makefile b/net/ipv4/netfilter/Makefile
index 4811159..a1eb4f6 100644
--- a/net/ipv4/netfilter/Makefile
+++ b/net/ipv4/netfilter/Makefile
@@ -71,3 +71,5 @@ obj-$(CONFIG_IP_NF_ARPFILTER) += arptable_filter.o
 
 obj-$(CONFIG_IP_NF_QUEUE) += ip_queue.o
 
+# Ignore warnings to prevent build breaks when using gcc 4.5.0
+CFLAGS_REMOVE_nf_nat_standalone.o = -Werror
diff --git a/net/netfilter/Makefile b/net/netfilter/Makefile
index 14e3a8f..ec20378 100644
--- a/net/netfilter/Makefile
+++ b/net/netfilter/Makefile
@@ -99,3 +99,6 @@ obj-$(CONFIG_NETFILTER_XT_MATCH_U32) += xt_u32.o
 
 # IPVS
 obj-$(CONFIG_IP_VS) += ipvs/
+
+# Ignore warnings to prevent build breaks when using gcc 4.5.0
+CFLAGS_REMOVE_nf_conntrack_netlink.o = -Werror

